app/jobs/list_project_contents_job.rb

```diff
@@ -21,12 +23,10 @@ def mediaflux_session
       logon_request.session_token
     end
 
-    # TODO: This only works with one server. In a multi-server environment
-    # there is no guarantee that the client will be able to fetch the file
-    # since requests could go to any of the many servers.
-    # See https://github.com/pulibrary/tiger-data-app/issues/523
     def filename_for_export
-      "#{Dir.pwd}/public/#{job_id}.csv"
+      raise "Shared location is not configured" if Rails.configuration.mediaflux["shared_files_location"].blank?
```

Comment:
This is the core of this PR. Files are now saved to a shared location so that they can be fetch by any web server when the user wants to download it.